src/controller/AutoCommissioner.cpp

```diff
@@ -117,7 +117,11 @@ CHIP_ERROR AutoCommissioner::SetCommissioningParameters(const CommissioningParam
         }
         else
         {
+#if defined(__GNUC__) && (__GNUC__ >= 12) // -Werror=dangling-pointer= observed with GNU 12.1.0
```

Comment:
Can we rather fix the issue rather than hide it? `auto & code = params.GetCountryCode().Value();` should be changed to `auto code = params.GetCountryCode().Value();` as `params.GetCountryCode()` returns a temporary object rather than a reference.

Reply:
Fixed

Reply:
Thanks!